src/cache/src/index.ts

```diff
@@ -347,36 +347,107 @@ export class Cache extends LRUCache<
       return this.integrityPath(i)
     } catch {}
   }
+
+  async #writeFileAtomic(file: string, data: Buffer | string) {
+    // ensure we get a different random key for every write,
+    // just in case the same file tries to write multiple times,
+    // it'll still be atomic.
+    const tmp = `${file}.${this.#random}${this.#i++}`
```

Comment:
```suggestion
    const tmp = `${file}.${this.#random}.${this.#i++}`
```

Reply:
Not a significant functional difference, it's just as unique without the `.`, but that delimiter makes it clearer from the filenames that this is the nth write for a given Cache instance, which has been handy for debugging a few times.